synapse/http/server.py

```diff
@@ -405,12 +405,14 @@ def respond_with_json(request, code, json_object, send_cors=False,
         return
 
     if pretty_print:
-        json_bytes = encode_pretty_printed_json(json_object) + "\n"
+        json_bytes = (encode_pretty_printed_json(json_object) + "\n"
+                      ).encode("utf-8")
     else:
         if canonical_json or synapse.events.USE_FROZEN_DICTS:
+            # canonicaljson already encodes to bytes
             json_bytes = encode_canonical_json(json_object)
         else:
-            json_bytes = simplejson.dumps(json_object)
+            json_bytes = simplejson.dumps(json_object).encode("utf-8")
```

Comment:
what, out of interest, is the logic behind using "utf-8" here? simplejson.dumps will produce ascii by default, so presumably any encoding (ascii, utf-8, latin-1?) will do the same thing here: I'd be keen that we use whatever is fastest.